Daksaray/8 symbol integration

daksaray opened the pull request:
Added .cc and .h files for number, index, object_color, and modified CMake file accordingly.

daksaray committed:
added number

daksaray committed:
added index

daksaray committed:
added object_color

daksaray committed:
Merge remote-tracking branch 'origin/master' into daksaray/8_symbol_integration

daksaray committed:
changed the format for index (after Tom's updates)

daksaray committed:
modified the format for number (after Tom's updates)

daksaray commented:
The second commit is based on the changes in number and index codes after Tom's updates. 

object_color codes have not been updated based on the new format yet (they were only moved from h2sl_nsf_nri_mvli to h2sl).

daksaray committed:
changed the format of object_color

daksaray committed:
changed format for spatial_relation

daksaray committed:
changed the format for object_property

daksaray committed:
changed the format for container

daksaray committed:
changed the format for abstract_container, but the code is not compiling now. we need to discuss whether we keep sub_symbols or properties (the same thing is also for object_property)

daksaray committed:
missing the symbols abstract_container, action, object_type, region_abstract_container, region_container

tmhoward commented on src/dcg/dcg.cc:
```diff
@@ -117,6 +126,75 @@ fill_search_spaces( const World* world ){
+    
+  /*Abstract symbols need to be initialized seperately in inference */
+  vector< std::string > container;
+  constraints.push_back( "group" );
```
switch "constraints" to "container"

tmhoward replied:
"group" can become "na" (need to check on corpus as well)

tmhoward commented on src/dcg/dcg.cc:
```diff
@@ -117,6 +126,75 @@ fill_search_spaces( const World* world ){
+  constraints.push_back( "na" );
+    
+  vector< std::string > number;
+  constraints.push_back( "na" );
```
switch "constraints" to "number"

tmhoward commented on src/dcg/dcg.cc:
```diff
@@ -117,6 +126,75 @@ fill_search_spaces( const World* world ){
   constraints.push_back( "outside" );
+    
+  vector< std::string > index;
+  constraints.push_back( "first" );
```
switch "constraints" to "index"

daksaray committed:
swicthed constraints to appropriate symbol names

rohanpaul committed:
object type symbol added

rohanpaul committed:
object type symbol properties added

rohanpaul committed:
abstract container

rohanpaul committed:
abstract container

daksaray committed:
multiple modifications: static/dynamic cast in container files, ostream format change, world file transfer (but not complete)

daksaray committed:
Merge branch 'daksaray/8_symbol_integration' of https://github.com/tmhoward/h2sl into daksaray/8_symbol_integration

abstract container changes merged

daksaray committed:
include all classes to grounding << operator and grounding_set from_xml function

daksaray committed:
removed commented out code

daksaray committed:
added abstract features to the list

rohanpaul committed:
header file guard error fixed

rohanpaul committed:
world file edits

rohanpaul committed:
world files temporary change

daksaray committed:
added feature_abstract_container_matches_child

daksaray committed:
transfered region_container symbols and features (feature_region_container_matches_child) and (feature_container_matches_child)

daksaray committed:
moved feature_action_object_matches_child but symbol action needs to be tranfered first, currently the changes are commented out

daksaray committed:
added feature_spatial_relation_matches_child

daksaray committed:
added feature_is_abstract_container

daksaray committed:
moved feature_is_object

daksaray committed:
move dfeature_is_container

daksaray committed:
added feature_is_region and feature_is_region_container

daksaray committed:
moved feature_is_region_abstract_container (commented out due to the absence of region_abstract_container symbol) and feature_is_spatial_relation

daksaray committed:
moved some features but compile errors, will look at later

daksaray committed:
added feature_container_type_matches_child_container_type

daksaray committed:
solve container problem (grounding is replaces with container in container.cc and container.g)

daksaray committed:
added feature_region_object_property_value to ostream

daksaray committed:
new feature added

rohanpaul committed:
dynamic cast of the grounding into the object and change from type() to object_type()

daksaray committed:
add feature_container_merge_container_pair(ongoing)

daksaray committed:
dynamic cast for type() error

daksaray committed:
added feature_phrase_has_pos_tag

daksaray committed:
added feature_phrase_has_single_pos_tag

daksaray committed:
added feature_phrase_has_ordered_pos_tag_pair

daksaray committed:
added feature_object_property_merge_object_property_spatial_relation

daksaray committed:
added feature_container_matches_empty_child_container

daksaray committed:
added feature_container_merge_empty_container_container

daksaray committed:
added feature_container_merge_object_property_container

daksaray committed:
added feature_container_merge_container_spatial_relation

daksaray committed:
added feature_region_container_merge_container_spatial_relation

daksaray committed:
added feature_region_container_container_matches_child_container

rohanpaul committed:
object property file names

rohanpaul committed:
obtaining upstream changes
Merge branch 'daksaray/8_symbol_integration' of https://github.com/tmhoward/h2sl into daksaray/8_symbol_integration

rohanpaul committed:
api consistency

daksaray committed:
to ease the feature transfer, the followings are replaced: object_type -> type, index_type->index, number_value->number, spatial_relation_type -> relation_type, container_type -> type, object_color_type -> color

daksaray committed:
moved feature_object_property_index (some compilation errors)

daksaray committed:
moving region_abstract_container symbol (not complete)

arkinj commented:
Is this pull request no longer desired because we now have the ADCG branch?

daksaray commented:
I believe so. Because Rohan has created the adcg branch based on daksaray/8_symbol_integration.
